ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -951,6 +962,21 @@ void Matmul::validate(
     const auto& input_tensor_b = input_tensors.at(1);
     const auto& a_shape = input_tensor_a.get_shape();
     const auto& b_shape = input_tensor_b.get_shape();
+    auto in0_tile_shape = input_tensor_a.get_tile().get_tile_shape();
```

Comment:
can we capture the return like this auto [in0_tile_height, in0_tile_width] = ....

then use in0_tile_height or in0_tile_width, makes the code much more readable.

Reply:
I wouldn't worry too much here, as we always use height then width.
take a look at ShardSpec in the file, where shape is also defined as` std::array<uint32_t, 2> shape;`
`tt_metal/impl/buffers/buffer.hpp`